### DISCUSSION TITLE
Suggestions for video recording feature

### DISCUSSION THREAD

**Discussion by s-trinh** [cyberbotics/webots#3899] [Ideas]
I would like to suggest the following ideas when using the `Make Movie` button in Webots:

- add an option to choose the folder where the images will be saved. We have a simulation project that can run for more or less 24 hours (simulation speed is ~0.3x), and on Ubuntu the images are saved in the `/tmp` folder that can run out of space with a SSD. It would be useful to be able to chose an external hard drive.
- maybe add an option to choose the video compression rate / switch to another compression method:
  - I think the current script use two passes compression:
  ```
  ffmpeg -loglevel warning -y -f image2 -r 25 -i "webotsFrame_14660_%06d.jpg" -b:v 18225000 -vcodec libx264 -pass 1 -g 132 -an -pix_fmt yuvj420p video.mp4
  ffmpeg -loglevel warning -y -f image2 -r 25 -i "webotsFrame_14660_%06d.jpg" -b:v 18225000 -vcodec libx264 -pass 2 -g 132 -an -pix_fmt yuvj420p video.mp4
  ```
  - with our long simulation, there are ~600000 images and the resulting video is quite big
  - what I am doing instead is to not click on the popup windows when it asks for the filename, and use my own ffmpeg command line to convert the images to a video
  - I suggest adding an option to choose the bitrate, and also to switch to [Constant Rate Factor (CRF) compression](https://trac.ffmpeg.org/wiki/Encode/H.264#crf), this is what I am using and the video file is smaller while preserving good image quality
- maybe add an option to choose the number of digits (`_%06d.jpg`), I don't know if it can be an issue with more than 999999 images, and when ffmpeg will glob the files?

**Comment by omichel** [comment]
That sounds all good. Please go ahead and proceed with the implementation in Webots. We will be happy to review your contribution in a new PR on this repo.
